src/App.tsx

```diff
@@ -14,46 +14,58 @@ import Account from "./pages/Account/Account";
 import UpdateProfile from "./pages/UpdateProfile/UpdateProfile";
 import { PrivateRoute } from "./components/PrivateRoute/PrivateRoute";
 import AdminDashboard from "./pages/AdminDashboard/AdminDashboard";
+import { useState, useEffect } from "react";
+import Loading from "./components/Loading/Loading";
 import Dashboard from "./pages/Dashboard/Dashboard";
 import Create from "./pages/Create/Create";
 
-
 const App = () => {
   const { isAuthenticated, isAdmin } = useAuth();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
```

Comment:
do we need this line, as we are already setting loading to true in the state by default

Reply:
Don't think so